ethcore/src/snapshot/mod.rs

```diff
@@ -281,11 +338,33 @@ pub fn chunk_state<'a>(db: &HashDB<KeccakHasher>, root: &H256, writer: &Mutex<Sn
 	let mut used_code = HashSet::new();
 
 	// account_key here is the address' hash.
-	for item in account_trie.iter()? {
+	let mut account_iter = account_trie.iter()?;
+
+	let mut seek_to = None;
+
+	if let Some(part) = part {
+		let part_offset = 256 / SNAPSHOT_SUBPARTS;
+		let mut seek_from = vec![0; 32];
+		seek_from[0] = (part * part_offset) as u8;
+		account_iter.seek(&seek_from)?;
+
+		// Set the upper-bond, except for the last part
+		if part < SNAPSHOT_SUBPARTS - 1 {
```

Comment:
we could get rid of this `if`, if we make `seek_to` inclusive upper bound, e.g.  the check would be
```rust
			if account_key[0] > seek_to {
				break;
			}
```

Reply:
But `seek_to` wouldn't make sense if `part == None`, so it might as well be an Option